### ISSUE TITLE
upgrading create-react-app from v1.5.1 to v2 cause error in yarn serve

### ISSUE THREAD

**Issue by manian-kumaran** [cereallarceny/cra-ssr#27]
upgrading create-react-app from v1.5.1 to v2 (https://github.com/facebook/create-react-app/issues/5103):

yarn serve fails

create-react-app team does not support SSR, so they can't fix it:
https://github.com/facebook/create-react-app/issues/5103#issuecomment-425696664

Upgrading from v1.1.5 to v2 using
npm install react-scripts@next --save

for existing (non-ejected) create-react-app:
https://github.com/cereallarceny/cra-ssr/
https://medium.com/@cereallarceny/server-side-rendering-in-create-react-app-with-all-the-goodies-without-ejecting-4c889d7db25e

This app has ssr in the server folder:
server/index.js
that contains:
....
require('babel-register')({
ignore: //(build|node_modules)//,
presets: ['env', 'react-app'],
plugins: [
'syntax-dynamic-import',
'dynamic-import-node',
'react-loadable/babel'
]
});

and package.json:
"scripts": {
"start": "react-scripts start",
"build": "GENERATE_SOURCEMAP=false react-scripts build",
"test": "react-scripts test --env=jsdom",
"precommit": "lint-staged",
"serve": "NODE_ENV=production node ./server/index.js",
"eject": "react-scripts eject"

After installing missing dependencies:
yarn add query-string
yarn add babel-plugin-syntax-dynamic-import
yarn add babel-plugin-dynamic-import-node

yarn build is successful
But
yarn serve
throws error:
Error: Couldn't find preset "env" relative to directory .....

If the 'env' is removed
from
presets: ['env', 'react-app'],

yarn serve
throws different error
Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing preset: .....

**Comment by JamesLeBoeuf** [comment]
Following this.

**Comment by ishakimanuel** [comment]
I am trying to fix this issue by using new babel and babel plugins version and remove "react-app" in list of presets.
`yarn add @babel/register @babel/polyfill @babel/babel-plugin-dynamic-import-node  --dev`

>```
> require("@babel/register")({
>     ignore: [/\/(build|node_modules)\//],
>     presets: ["@babel/env", "@babel/preset-react"],
>     plugins: ["@babel/plugin-syntax-dynamic-import", "dynamic-import-node", "react-loadable/babel"]
> });
>
> require("@babel/polyfill");
>
> ```

**Comment by cereallarceny** [comment]
This requires the project to upgrade the build system to Babel 7.  If anyone wants to take this on - go for it.  I'll merge any working PR.

Adding the `hacktoberfest` label now.  :)

**Comment by arturparkhisenko** [comment]
@ishakimanuel i did that yesterday and got another issue in tests, without tests it's working.
This is what need to be changed in the `package.json`:
```js
"dependencies": {
  ...
  + "query-string": "^6.2.0",
  ~ "react-scripts": "^2.0.3",
  ...
},
"scripts": {
  ...
  ~ "test": "react-scripts test",
  ...
},
// Also it'll ask you to include default browserslist config, it'll be generated automatically
```
in the `index.js` of server:
```
// ...
// old require('babel-register')({
require('@babel/register')({
  ignore: /\/(build|node_modules)\//,
  // old: presets: ['env', 'react-app'],
  presets: ['@babel/preset-env', 'react-app'],
  plugins: [
    // old: 'syntax-dynamic-import',
    '@babel/plugin-syntax-dynamic-import',
    'dynamic-import-node',
    'react-loadable/babel'
  ]
});
```

Tests error:
```
Error: Uncaught [Invariant Violation: You should not use <Route> or withRouter() outside a <Router>]
```

**Comment by arturparkhisenko** [comment]
But i've tested master too, looks like it's not related to my changes, tests are broken for a long time)

**Comment by arturparkhisenko** [comment]
I think that we should export/import components for tests differently. And then it'll start working.

**Comment by cereallarceny** [comment]
Closed by #30

**Comment by heshamelmasry77** [comment]
i am getting this error :
```
internal/modules/cjs/loader.js:582
    throw err;
    ^

Error: Cannot find module '@babel/register'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
    at Function.Module._load (internal/modules/cjs/loader.js:506:25)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/hesham/Documents/GitHub/room-me/server/index.js:30:1)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```
